neuror/cut_plane/legacy_detection.py

```diff
@@ -4,30 +4,73 @@
 https://bbpcode.epfl.ch/source/xref/platform/BlueRepairSDK/BlueRepairSDK/src/repair.cpp#263
 '''
 import logging
-import numpy as np
+from collections import defaultdict
 
+import numpy as np
+from morph_tool import apical_point_section_segment
 from neurom import iter_sections
 from neurom.core import Tree
 from neurom.core.dataformat import COLS
 
+from neuror.utils import RepairType, repair_type_map
+
 L = logging.getLogger(__name__)
 
 
-def cut_detect(neuron, axis):
+def children_ids(section):
+    '''
+    https://bbpcode.epfl.ch/source/xref/platform/BlueRepairSDK/BlueRepairSDK/src/helper_dendrite.cpp#111
+
+    Except that it returns section instead of ids
+
+    '''
+    return list(section.ipreorder())
+
+
+def internal_cut_detection(neuron, axis):
+    '''As in:
+
+    https://bbpcode.epfl.ch/source/xref/platform/BlueRepairSDK/BlueRepairSDK/src/repair.cpp#263
+
+'''
+    axis = {'x': COLS.X, 'y': COLS.Y, 'z': COLS.Z}[axis.lower()]
+
+    cut = defaultdict(lambda key: False)
+    side = cut_detect(neuron, cut, 0, axis)
+
+    # reclassify cut points in tuft,based on apical point position
+    apical_section_id, point_id = apical_point_section_segment(neuron)
+    if apical_section_id is not None:
+        apical_section = neuron.sections[apical_section_id]
+        apical_offset = apical_section.points[point_id, axis]
+        cut_mark(children_ids(apical_section), cut, apical_offset, side, axis)
+    else:
+        apical_section = None
+
+    extended_types = repair_type_map(neuron, apical_section)
+    oblique_roots = get_obliques(neuron, extended_types)
+
+    # reclassify points in obliques. based on the position of their root.
+    for root in oblique_roots:
+        offset = root.points[0]
+
+        # z is hard coded in the original code as well
+        cut_mark(root.children, cut, offset[COLS.Z], side, axis)
+
+    cut_leaves = np.array([sec.points[-1, COLS.XYZ] for sec, is_cut in cut.items() if is_cut])
+
+    return cut_leaves, side
+
+
+def cut_detect(neuron, cut, offset, axis):
```

Comment:
`cut` is implicitly initialised in this function. Should we leave it as it was before?

Reply:
I meant that `cut` is initialized outside:
```
cut = defaultdict(lambda key: False)
```
and after it is filled in the function `cut_detect` implicitly. I consider it implicit because `cut_detect` does not return `cut` but takes it only as an argument.
Anyway, I see that you are aware of all this, and this change is rather a requirement to have an exactly the old behaviour. Whoever asked for this, all the responsibility to him.